File: WealthBuilder/frontend/src/services/dashboardService.ts
import { apiRequest } from './apiClient';
import { DASHBOARD_ENDPOINTS } from '../constants/api';
import type { AssetDistribution } from '../types/dashboard';


// Read-only home-screen aggregations for the signed-in user.

export const fetchDistribution = (): Promise<AssetDistribution[]> => {
    return apiRequest<AssetDistribution[]>(DASHBOARD_ENDPOINTS.DISTRIBUTION);
};

File: WealthBuilder/backend/src/main/java/com/wealthbuilder/backend/services/interfaces/DashboardService.java
package com.wealthbuilder.backend.services.interfaces;

import com.wealthbuilder.backend.dtos.dashboard.AssetDistributionResponse;

import java.util.List;


/**
 * Read-only dashboard aggregations for the authenticated user's home screen.
 */
public interface DashboardService {

    /**
     * Net invested per asset for the donut, highest first; assets the user holds nothing in
     * are omitted.
     */
    List<AssetDistributionResponse> distribution(String username);
}
